libraries/app/api.cpp

```diff
@@ -302,14 +302,14 @@ namespace graphene { namespace app {
        const auto& db = *_app.chain_database();
        FC_ASSERT( limit <= 100 );
        vector<operation_history_object> result;
-       account_statistics_object stats;
-       try { stats = account(db).statistics(db); } catch(...) { return result; }
-       const account_transaction_history_object* node;
-       try { node = &stats.most_recent_op(db); } catch(...) { return result; }
-       if( start == operation_history_id_type() )
-          start = node->operation_id;
-       if( start.instance.value > node->operation_id.instance.value )
-          start = node->operation_id;
+       try {
+          optional<account_statistics_object> stats;
+          optional<account_transaction_history_object> node;
+          stats = account(db).statistics(db);
+          node = stats->most_recent_op(db);
```

Comment:
Not beautiful enough.
* No need to use `optional` now since all needed variables are in one code block. And
* better use `const &`.

E.G.

```
const account_transaction_history_object& node = account(db).statistics(db).most_recent_op(db);
```

Although so many `(db)` here looks a bit odd.